scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        values is too coarse to represent an argument for which the log-CDF is
+        close to the provided `x`, even when the function is theoretically
+        continuous within the domain of the argument.
+
+        >>> X.ilogcdf(-np.inf), X.ilogcdf(-1e10), X.ilogcdf(-30.)
+        (-0.5, nan, -0.4999999999998863)
+
+        """
+        return self._ilogcdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _ilogcdf_dispatch(self, x, method=None, **kwargs):
+        if self._overrides('_ilogcdf_formula'):
+            method = self._ilogcdf_formula
+        elif self._overrides('_ilogccdf_formula'):
+            method = self._ilogcdf_complementarity
+        else:
+            method = self._ilogcdf_inversion
+        return method
+
+    def _ilogcdf_formula(self, x, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _ilogcdf_complementarity(self, x, **kwargs):
+        return self._ilogccdf_dispatch(_log1mexp(x), **kwargs)
+
+    def _ilogcdf_inversion(self, x, **kwargs):
+        return self._solve_bounded(self._logcdf_dispatch, x, kwargs=kwargs)
+
+    @_set_invalid_nan
+    def icdf(self, x, *, method=None):
+        r"""Inverse cumulative distribution function
+
+        The inverse cumulative distribution function :math:`F^{-1}(x)` is the
+        argument :math:`y` for which the cumulative distribution function
+        :math:`F(y)` evaluates to :math:`x`.
+
+        .. math::
+
+            F^{-1}(x) = y \text{ such that } F(y) = x`
+
+        ``icdf`` accepts `x` for :math:`x \in [0, 1]`.
+
+        Parameters
+        ----------
+        x : Array
+            The argument of the inverse cumulative distribution function
+            (inverse CDF).
+        method : {None, 'formula', 'complementarity', 'inversion'}
+            The strategy used to evaluate the inverse CDF.
+            By default (``None``), the infrastructure chooses between the
+            following options.
+
+            - ``'formula'``: use a formula for the inverse CDF itself
+            - ``'complementarity'``: evaluate the inverse CCDF at the
+                                     complement of `x`; i.e. ``1 - x``.
+            - ``'inversion'``: solve numerically for the argument at which the
+                               CDF is equal to `x`.
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The inverse CDF evaluated at the provided argument.
+
+        See Also
+        --------
+        ContinuousDistribution.cdf
+        ContinuousDistribution.ilogcdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`. The
+        inverse CDF takes on its minimum value of :math:`l` at :math:`x = 0`
```

Comment:
I don't understand. "Takes on" is what appear on this line, and it is what I intended.

Reply:
So it's ESL. The only meaning of "take on " I knew was "to be ready and willing to meet an adversary or opponent". Now I know the other meaning, so please ignore this and the other comment (amusingly, GH does not allow me to resolve comments here, maybe the file is too large).

Reply:
I changed it to "returns" because it is more appropriate to use "takes on" to refer to when a random variable is observed to be a particular value.
I'll resolve this for you, then.